plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftConnectorTest.java

```diff
@@ -116,6 +152,60 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
         };
     }
 
+    private RedshiftSplitManager createSplitManager()
+    {
+        DriverConnectionFactory driverConnectionFactory = DriverConnectionFactory.builder(new Driver(), JDBC_URL, new ExtraCredentialProvider(Optional.empty(), Optional.empty(), new StaticCredentialProvider(Optional.of(JDBC_USER), Optional.of(JDBC_PASSWORD))))
+                .build();
+        RedshiftClient redshiftClient = new RedshiftClient(
+                new BaseJdbcConfig().setConnectionUrl(JDBC_URL),
+                redshiftJdbcConfig,
+                driverConnectionFactory,
+                new JdbcStatisticsConfig(),
+                new DefaultQueryBuilder(NONE),
+                new DefaultIdentifierMapping(),
+                NONE);
+
+        return new RedshiftSplitManager(
+                redshiftClient,
+                new DefaultQueryBuilder(NONE),
+                NONE,
+                new JdbcSplitManager(redshiftClient),
+                redshiftJdbcConfig,
+                new S3FileSystemFactory(OpenTelemetry.noop(), new S3FileSystemConfig()
+                        .setAwsAccessKey(AWS_ACCESS_KEY)
+                        .setAwsSecretKey(AWS_SECRET_KEY)
+                        .setRegion(AWS_REGION)
+                        .setStreamingPartSize(DataSize.valueOf("5.5MB")), new S3FileSystemStats()),
+                Executors.newFixedThreadPool(1));
+    }
+
+    @Test
+    void testJdbcFlow()
```

Comment:
This test should also be based on inspecting query stats rather than using a custom splits manager.
If really needed, you should create a separate unit test class `TestRedshiftSplitManager` and have only query runner tests here.

Reply:
Sorry I missed it earlier.